Source/SwiftLintFramework/Models/Configuration.swift

```diff
@@ -32,9 +32,47 @@ public struct Configuration: Equatable {
     public let rules: [Rule]
     public var rootPath: String?              // the root path to search for nested configurations
     public var configurationPath: String?     // if successfully loaded from a path
-    public var hash: Int?
     public let cachePath: String?
 
+    internal var cacheDescription: String {
+        let cacheRulesDescriptions: [String: Any] = rules.reduce([:]) { accu, element in
+            var accu = accu
+            accu[type(of: element).description.identifier] = element.cacheDescription
+            return accu
+        }
+        let dict: [String: Any] = [
+            "root": rootPath ?? FileManager.default.currentDirectoryPath,
+            "rules": cacheRulesDescriptions
+        ]
+        if let jsonData = try? JSONSerialization.data(withJSONObject: dict),
```

Comment:
I tried this PR on Linux(norionomura/sourcekit:311).
[`cache.json`](https://gist.github.com/norio-nomura/e3584c423ae0ba61197c0bb542d7d629) contains three variants of `configurationDescription` after three times executing `.build/debug/swiftlint lint --cache-path .` in SwiftLint working copy.